samza-core/src/main/java/org/apache/samza/clustermanager/ContainerAllocator.java

```diff
@@ -221,7 +221,10 @@ void assignResourceRequests() {
           updateExpiryMetrics(request);
           if (hostAffinityEnabled) {
             handleExpiredRequestWithHostAffinityEnabled(processorId, preferredHost, request);
+          } else {
+            handleExpiredRequestWithHostAffinityDisabled(processorId, request);
```

Comment:
It might be better to have a single method called handleExpiredRequest(processorId, request, preferredHost). Perhaps along the lines: 
`void handleExpiredRequest(String processorId, String preferredHost, SamzaResourceRequest request) {

    boolean resourceAvailableOnAnyHost = hasAllocatedResource(ResourceRequestState.ANY_HOST);
    if (hostAffinityEnabled && standbyContainerManager.isPresent()) {
      standbyContainerManager.get()
          .handleExpiredResourceRequest(processorId, request,
              Optional.ofNullable(peekAllocatedResource(ResourceRequestState.ANY_HOST)), this, resourceRequestState);
    } else if (resourceAvailableOnAnyHost) {
      LOG.info("Request for Processor ID: {} on host: {} has expired. Running on ANY_HOST", processorId, preferredHost);
      runStreamProcessor(request, ResourceRequestState.ANY_HOST);
    } else {
      LOG.info("Request for Processor ID: {} on host: {} has expired. Requesting additional resources on ANY_HOST.",
          processorId, preferredHost);
      resourceRequestState.cancelResourceRequest(request);
      requestResource(processorId, ResourceRequestState.ANY_HOST);
    }
  }`